kura/org.eclipse.kura.ble.provider/src/main/java/org/eclipse/kura/internal/ble/util/BluetoothProcess.java

```diff
@@ -106,17 +110,30 @@ void execSnoop(String[] cmdArray, final BTSnoopListener listener) {
         }
         Consumer<CommandStatus> callback = status -> logger.debug("Command ended with exit value {}",
                 status.getExitStatus().getExitCode());
-        Command command = new Command(cmdArray);
+
+        // Build command
+        List<String> commandLine = new ArrayList<>();
+        commandLine.add("{");
+        commandLine.add("exec");
+        for (int i = 0; i < cmdArray.length; i++) {
+            commandLine.add(cmdArray[i]);
+        }
+        commandLine.add(">/dev/null;");
+        commandLine.add("}");
+        commandLine.add("3>&1");
+
+        Command command = new Command(commandLine.toArray(new String[commandLine.size()]));
         command.setOutputStream(this.outputStream);
         command.setErrorStream(this.errorStream);
+        command.setExecuteInAShell(true);
         this.executorService.execute(command, callback);
 
         this.futureInputGobbler = streamGobblers.submit(() -> {
             Thread.currentThread().setName("BluetoothProcess BTSnoop Gobbler");
             try {
                 readBTSnoopStreamFully(this.readOutputStream, listener);
             } catch (IOException e) {
-                logger.error("Process snoop input stream failed", e);
+                logger.debug("Process snoop input stream failed", e);
             }
```

Comment:
Why `debug`?

Reply:
Since the introduction of the CommandExecutorService the btdump process is killed independent of the btsnoop parsing which means the stream almost always ends with an incomplete btsnoop packet. The btsnoop parser doesn't check for available bytes before attempting to read which is fair enough, but does mean this exception is thrown more often than not when the btdump process ends which is an expected error. This does have the potential to hide more serious issues parsing the btsnoop data but unless the BluetoothProcess handles the btdump process directly like it used to this isn't easily avoided.